src/main/java/ua/com/javarush/oleksandr/reddit/redditcloneabstract/service/VoteService.java

```diff
@@ -0,0 +1,33 @@
+package ua.com.javarush.oleksandr.reddit.redditcloneabstract.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import ua.com.javarush.oleksandr.reddit.redditcloneabstract.dto.VoteDTO;
+import ua.com.javarush.oleksandr.reddit.redditcloneabstract.model.Vote;
+import ua.com.javarush.oleksandr.reddit.redditcloneabstract.repository.PostRepository;
+import ua.com.javarush.oleksandr.reddit.redditcloneabstract.repository.UserRepository;
+import ua.com.javarush.oleksandr.reddit.redditcloneabstract.repository.VoteRepository;
+
+@Service
+@RequiredArgsConstructor
+public class VoteService {
+
+    private final VoteRepository voteRepository;
+    private final UserRepository userRepository;
+    private final PostRepository postRepository;
+
+    @Transactional
+    public void vote(VoteDTO voteDto) {
+        boolean voteExists = voteRepository.existsByUserUserIdAndPostId(voteDto.getUserId(), voteDto.getPostId());
+        if (voteExists) {
```

Comment:
What if we want to change our vote decision?)

Reply:
thank you for the apt remark, I will definitely work on this option